### PR TITLE
adding rabbitmq docker image to  compose file 

### PR DESCRIPTION
**Pull request by marwanehcine**
To introduce rabbitmq message broker in georchestra infrastructure, we add docker image to compose file

### CODE REVIEW

**Comment by jeanmi151** [comment]
hey thanks @marwanehcine for this PR, 
could you add healthcheck, dependancies with others service, and maybe ressources limitation 
thanks

**Comment by fvanderbiest** [comment]
> could you add healthcheck, dependancies with others service, and maybe ressources limitation

secrets too.
Maybe Marwane needs some help from a devops to do this ?

File: docker-compose.yml [hunk 1]
```diff
@@ -390,3 +392,20 @@ services:
       JAVA_OPTS: -Dfile.encoding=UTF-8
     volumes:
       - georchestra_datadir:/etc/georchestra
+      
+  rabbitmq:
+    image: docker.io/bitnami/rabbitmq:3.12
+    ports:
+      - '4369:4369'
```
Comment (edevosc2c) [inline docker-compose.yml]: do we really need to expose by default any rabbitmq ports? the applications that are going to use rabbitmq are already inside docker, exposing sensitive ports is not a good idea security wise.

File: docker-compose.yml [hunk 2]
```diff
@@ -14,6 +14,8 @@ volumes:
   datafeeder_postgis_data:
   esdata:
   georchestra_datadir:
+  rabbitmq_data:
+    driver: local
```
Comment (edevosc2c) [inline docker-compose.yml]: this is unnecessary, the driver is already local by default

**Review by pmauduit — APPROVED** [review]

File: docker-compose.yml [hunk 3]
```diff
@@ -390,3 +392,12 @@ services:
       JAVA_OPTS: -Dfile.encoding=UTF-8
     volumes:
       - georchestra_datadir:/etc/georchestra
+      
+  rabbitmq:
+    image: docker.io/bitnami/rabbitmq:3.12
+    environment:
```
Comment (fvanderbiest) [inline docker-compose.yml:406]: why not using the env file here ?

File: .envs-rabbitmq [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#envs-rabbitmq
+RABBITMQHOST=rabbitmq
```
Comment (jeanmi151) [inline .envs-rabbitmq]: this var should not be here !
but in here : https://github.com/georchestra/docker/blob/master/.envs-hosts

File: .envs-rabbitmq [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# envs-rabbitmq
+RABBITMQ_USERNAME=georchestra
+RABBITMQ_PASSWORD=georchestra
```
Comment (fvanderbiest) [inline .envs-rabbitmq:3]: Shouldn't we have a new secret for RABBITMQ_PASSWORD ?

Comment (edevosc2c) [reply .envs-rabbitmq:3]: I don't get it, `envs` files are technically "secrets" because there is no concept of secrets inside docker.

Comment (fvanderbiest) [reply .envs-rabbitmq:3]: I was refering to https://github.com/georchestra/docker/blob/6d084eb9670f54cc738d63489c9f2915eaf27524/docker-compose.yml#L18-L22

Comment (edevosc2c) [reply .envs-rabbitmq:3]: oh ok, for that to work the container/application has to support the ability to read the password **from a file** (**not from an environment variable**).

I did look at the documentation of the container that wasn't explained anywhere BUT I did an extensive search and found this (for a related container maintained by bitnami): https://github.com/bitnami/containers/issues/46152

So yes we can use RABBITMQ_PASSWORD_FILE for a password located in file.

Comment (fvanderbiest) [reply .envs-rabbitmq:3]: Excellent, please go ahead.

Comment (edevosc2c) [reply .envs-rabbitmq:3]: The only issue I see when doing this is that, the applications that use rabbitmq like the console or gateway probably do not support secrets using a file.
So if we implement it with a secret in a file, the user will have to both change the environment variable RABBITMQ_PASSWORD **AND** the file rabbitmq_password.txt located in the `secrets` folder.

Due to that I'm in favor of keeping the usage in an environment variable even though it's less secure.

**Comment by marwanehcine** [comment]
> hey thanks @marwanehcine for this PR, could you add healthcheck, dependancies with others service, and maybe ressources limitation thanks

healthcheck added, but dependencies , we need to check if we will make gateway or console dependent to rabbitmq, should be discussed in group.

**Comment by marwanehcine** [comment]
> hey thanks @marwanehcine for this PR, could you add healthcheck, dependancies with others service, and maybe ressources limitation thanks

healthcheck added, but making console or gateway depending on rabbitmq, we need to discuss it in group i think!

File: docker-compose.yml [hunk 6]
```diff
@@ -390,3 +393,19 @@ services:
...
+    image: docker.io/bitnami/rabbitmq:3.12
+    healthcheck:
+      test: ["CMD-SHELL", "curl -f --user georchestra:georchestra localhost:15672/api/health/checks/local-alarms >/dev/null || exit 1"]
+      interval: 30s
+      timeout: 10s
+      retries: 10
+    environment:
+      - RABBITMQ_LOGS=-
+      - RABBITMQ_USERNAME=georchestra
+      - RABBITMQ_PASSWORD=georchestra
```
Comment (jeanmi151) [inline docker-compose.yml]: better to add file for this two var
.envs-rabbitmq

File: docker-compose.yml [hunk 7]
```diff
@@ -390,3 +393,19 @@ services:
...
+    ports:
+      - '15672:15672'
```
Comment (jeanmi151) [inline docker-compose.yml]: do we really need/want that ?

File: .envs-rabbitmq (hunk 5 above)
Comment (fvanderbiest) [reply .envs-rabbitmq:3]: Sound fair to me. Thanks !

File: docker-compose.yml (hunk 6 above)
Comment (edevosc2c) [reply docker-compose.yml]: I have reverted the change done by marwane, that was in a env file before.

File: docker-compose.yml (hunk 7 above)
Comment (edevosc2c) [reply docker-compose.yml]: Like discussed in https://github.com/georchestra/docker/pull/213#discussion_r1304220744, we do not want to expose rabbitmq ports to the outside world.

**Pull request merged**